### PR TITLE
feat(retry): enhance retry logging for operations via storage-client

### PR DESCRIPTION
**Pull request by raj-prince**
### Description
- Using the the custom retrier function which provides more context about the retry, like, operation_name, attempt, retry_id (to trace all the retry for a given GCS api call). Integrate the changes for PR: https://github.com/googleapis/google-cloud-go/pull/13750
- Command used to trigger retry: `./gcsfuse --custom-endpoint="http://localhost:9999" --log-file=gcsfuse.log --log-format=text test-bucket mnt_test`
- Invocation-id can be used to collect all the retry logs for a particular invocation from GCSFuse.

**Before change:**
```
time="17/06/2026 01:04:41.376848" severity=WARNING message="Retrying for the error: Get \"http://localhost:9999/b/test-bucket/o?alt=json&delimiter=%2F&endOffset=&fields=nextPageToken%2Cprefixes%2Citems%28generation%2Cmetageneration%2Cupdated%2Cmetadata%2CcontentEncoding%2Ccrc32c%2Cname%2Csize%29&includeFoldersAsPrefixes=false&includeTrailingDelimiter=true&matchGlob=&maxResults=5000&pageToken=&prefix=&prettyPrint=false&projection=noAcl&startOffset=&versions=false\": dial tcp 127.0.0.1:9999: connect: connection refused" mount-id=test-bucket-668a92ca
```

**After change:**
```
time="17/06/2026 12:59:52.354686" severity=WARNING message="Retrying for transient error: Get \"http://localhost:9999/b/test-bucket/o?alt=json&delimiter=%2F&endOffset=&fields=nextPageToken%2Cprefixes%2Citems%28contentEncoding%2Ccrc32c%2Cname%2Csize%2Cgeneration%2Cmetageneration%2Cupdated%2Cmetadata%29&includeFoldersAsPrefixes=false&includeTrailingDelimiter=true&matchGlob=&maxResults=5000&pageToken=&prefix=&prettyPrint=false&projection=noAcl&startOffset=&versions=false\": dial tcp 127.0.0.1:9999: connect: connection refused, InvocationID: 1ef8ca0f-89a1-4b87-b4f5-183a4d38881d, Attempt: 6, Op: ListObjects, Object: \"\"" mount-id=test-bucket-236e2d01
```

### Testing details
1. Manual - tested with custom-endpoint.
2. Unit tests - added.
3. Integration tests - NA

### CODE REVIEW

File: internal/storage/storageutil/custom_retry_test.go [hunk 1]
```diff
@@ -281,14 +282,47 @@ func TestShouldRetryLogsWarning(t *testing.T) {
...
+		Object:       "some/file.txt",
+	}
+
+	// Act
+	retry := ShouldRetryWithRetryContext(err401, retryCtx)
+
+	// Assert
+	assert.True(t, retry)
+	logMsg := buf.String()
+	t.Logf("Captured Log Message:\n%s", logMsg)
```
Comment (meet2mky) [inline internal/storage/storageutil/custom_retry_test.go]: Nit: this log is not required for tests.
```suggestion
```

File: internal/storage/storageutil/custom_retry_test.go [hunk 2]
```diff
@@ -281,14 +282,47 @@ func TestShouldRetryLogsWarning(t *testing.T) {
...
-	retry := ShouldRetry(err401)
+	retry := ShouldRetryWithRetryContext(err401, nil)
 
 	// Assert
 	assert.True(t, retry)
 	assert.Contains(t, buf.String(), "WARNING")
 	assert.Contains(t, buf.String(), "Retrying for error-code 401")
 }
 
+func TestShouldRetryLogsWarningWithRetryContext(t *testing.T) {
```
Comment (meet2mky) [inline internal/storage/storageutil/custom_retry_test.go:293]: Can we add a similar negative test where the RetryContext is nil

**Review by meet2mky — DISMISSED** [review]
Looks good few test related comments.

**Review by meet2mky — APPROVED** [review]

**Review by kislaykishore — APPROVED** [review]
